File: frontend/src/api/task.ts
import api from './axios'

export interface Task {
  id: number
  projectId: number
  sprintId: number
  taskKey: string
  title: string
  description: string
  type: string
  status: string
  priority: string
  storyPoints: number
  originalEstimate: number
  remainingEstimate: number
  timeSpent: number
  assigneeId: number
  reporterId: number
  createdAt: string
}

export interface CreateTaskRequest {
  projectId: number
  sprintId?: number
  title: string
  description?: string
  type: string
  priority?: string
  storyPoints?: number
  originalEstimate?: number
  assigneeId?: number
}

export const taskApi = {
  getBySprint: (sprintId: number) => api.get<Task[]>(`/tasks/sprint/${sprintId}`),
  getByProject: (projectId: number) => api.get<Task[]>(`/tasks/project/${projectId}`),
  getById: (id: number) => api.get<Task>(`/tasks/${id}`),
  create: (data: CreateTaskRequest) => api.post<Task>('/tasks', data),
  update: (id: number, data: CreateTaskRequest) => api.put<Task>(`/tasks/${id}`, data),
  updateStatus: (id: number, status: string) => api.patch<Task>(`/tasks/${id}/status?status=${status}`),
}

export const createTask = async (sprintId: number, data: Omit<CreateTaskRequest, 'sprintId'>) => {
  const response = await taskApi.create({ ...data, sprintId })
  return response.data
}

File: backend/src/main/java/com/burndown/entity/Task.java
package com.burndown.entity;

import com.pgvector.PGvector;
import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.type.SqlTypes;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "tasks")
@EntityListeners(AuditingEntityListener.class)
public class Task {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "project_id", nullable = false)
    private Long projectId;

    @Column(name = "sprint_id")
    private Long sprintId;

    @Column(name = "parent_id")
    private Long parentId;

    @Column(name = "task_key", nullable = false, unique = true, length = 50)
    private String taskKey;

    @Column(nullable = false, length = 500)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private TaskType type;

    @Enumerated(EnumType.STRING)
    @Column(length = 20)
    private TaskStatus status = TaskStatus.TODO;

    @Enumerated(EnumType.STRING)
    @Column(length = 20)
    private TaskPriority priority = TaskPriority.MEDIUM;

    @Column(name = "story_points", precision = 5, scale = 1)
    private BigDecimal storyPoints;

    @Column(name = "original_estimate", precision = 10, scale = 2)
    private BigDecimal originalEstimate;

    @Column(name = "remaining_estimate", precision = 10, scale = 2)
    private BigDecimal remainingEstimate;

    @Column(name = "time_spent", precision = 10, scale = 2)
    private BigDecimal timeSpent = BigDecimal.ZERO;

    @Column(name = "assignee_id")
    private Long assigneeId;

    @Column(name = "reporter_id", nullable = false)
    private Long reporterId;

    @JdbcTypeCode(SqlTypes.ARRAY)
    @Column(columnDefinition = "TEXT[]")
    private String[] labels;

    @JdbcTypeCode(SqlTypes.JSON)
    @Column(name = "custom_fields", columnDefinition = "jsonb")
    private String customFields = "{}";

    @Column(name = "embedding", columnDefinition = "vector(1536)")
    private PGvector embedding;

    @CreatedDate
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @LastModifiedDate
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @Column(name = "resolved_at")
    private LocalDateTime resolvedAt;

    @Column(name = "due_date")
    private LocalDate dueDate;

    public enum TaskType {
        STORY, BUG, TASK, EPIC
    }

    public enum TaskStatus {
        TODO, IN_PROGRESS, IN_REVIEW, DONE, BLOCKED
    }

    public enum TaskPriority {
        CRITICAL, HIGH, MEDIUM, LOW
    }
}
